src/App.js

```diff
@@ -1,25 +1,71 @@
 import React from 'react';
-
 import './App.scss';
+import { Clock } from './clock';
+
+let interval;
+
+class App extends React.Component {
+  state = {
+    isClockVisible: false,
+    time: new Date().toLocaleTimeString(),
+    clockName: 0,
+  }
+
+  componentDidUpdate(_, prevState) {
+    clearInterval(interval);
+    const { isClockVisible, clockName } = this.state;
+
+    if (clockName !== prevState.clockName) {
+      // eslint-disable-next-line
+      console.log(`The Clock was renamed from ${prevState.clockName} to ${clockName}`)
+    }
+
+    if (isClockVisible) {
+      interval = setInterval(() => {
+        const date = new Date().toLocaleTimeString();
+
+        // eslint-disable-next-line
+        console.log(date);
+        this.setState({ time: date });
+      }, 1000);
+    }
+  }
```

Comment:
ComponentDidUpdate should be in Clock component, where you check previous and current props name, if it changed - print it to console.

Reply:
staren'kyi, it's much better this way, clock ne lagaet 💪